apps/server/src/stores/cachingStore.ts

```diff
@@ -0,0 +1,52 @@
+interface CacheData {
+  data: unknown;
+}
+
+const runtimeCache: Map<string, CacheData> = new Map();
+
+export function checkCached(key: string): boolean {
+  return Object.hasOwn(runtimeCache, key);
+}
+
+export function getCached<T>(key: string, callback: () => T): T {
+  if (!Object.hasOwn(runtimeCache, key)) {
+    try {
+      const data = callback();
+      runtimeCache.set(key, { data });
+    } catch (error) {
+      console.log(`Failed retrieving data from callback: ${error}`);
+    }
+  }
+
+  return runtimeCache.get(key).data as T;
+}
+
+export function setCached<T>(key: string, value: T): T {
+  try {
+    runtimeCache.set(key, { data: value });
+  } catch (error) {
+    console.log(`Failed retrieving data from callback: ${error}`);
```

Comment:
Error here doesn't make sense. I don't see a callback like on line 14

Reply:
the advantage of throwing an error would be to make sure the services do error handling around caching. not sure if this is more work than worth it?

Reply:
Woops, I meant the error message! Should've been more explicit 😅

Reply:
a-ha, great observation! This was definitely an unchecked copy-paste artifact. I guess set will never fail